File: lc1/solution.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_with_indices = [(nums[i], i) for i in range(len(nums))]
        nums_with_indices.sort(key = lambda x: x[0])

        left, right = 0, len(nums_with_indices) - 1
        while left < right:
            if (nums_with_indices[left][0] + nums_with_indices[right][0]) == target:
                break
            elif (nums_with_indices[left][0] + nums_with_indices[right][0]) > target:
                right -= 1
            else:
                left += 1

        return [nums_with_indices[left][1], nums_with_indices[right][1]]

File: lc1/Solution.java
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        List<Map.Entry<Integer, Integer>> numsWithIndices = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            numsWithIndices.add(Map.entry(nums[i], i));
        }
        numsWithIndices.sort(Map.Entry.comparingByKey());
        int left = 0;
        int right = numsWithIndices.size() - 1;
        while (left < right) {
            if ((numsWithIndices.get(left).getKey() + numsWithIndices.get(right).getKey() == target)) {
                break;
            } else if ((numsWithIndices.get(left).getKey() + numsWithIndices.get(right).getKey() > target)) {
                right--;
            } else {
                left++;
            }
        }

        return new int[] {left, right};
    }
}